proxy-node-shared/src/main/java/uk/gov/ida/notification/exceptions/mappers/VerifyServiceProviderResponseExceptionMapper.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.ida.notification.exceptions.mappers;
+
+import uk.gov.ida.notification.exceptions.proxy.VerifyServiceProviderResponseException;
+import uk.gov.ida.notification.views.ErrorPageView;
+
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import java.util.logging.Logger;
+
+public class VerifyServiceProviderResponseExceptionMapper implements ExceptionMapper<VerifyServiceProviderResponseException> {
```

Comment:
I think we can move this mapper to Gateway since no other component will use it as it returns a PageView. We can have another mapper for VSP errors in the Translator that just returns a 500 with an error message to the Gateway to handle (i.e. show an error page)

Reply:
I've made the exception and mapper more specific, so only the generate authn endpoint is handled and returned as html. This leaves the translate endpoint free to be mapped in a different way. 
I've had to leave the mapper and exception in shared however. The proxy is part of the shared package which has no access to stuff in the gateway, as it's not a dependency.

Reply:
That's fine 👍